lib/automatic/api_client/client_delegator.rb

```diff
@@ -0,0 +1,21 @@
+module Automatic
+  class ApiClient
+    class ClientDelegator
+      attr_reader :api_client, :delegated_class
+
+      def initialize(api_client, delegated_class)
+        @api_client = api_client
+        @delegated_class = delegated_class
+      end
+
+      def method_missing(method_sym, *arguments, &block)
+        super unless delegated_class.respond_to?(method_sym)
+
+        arguments[1 + delegated_class.method(method_sym).arity] ||= {}
```

Comment:
This looks pretty hairy. Hard to follow what it's doing this.

- take the incoming arguments (assuming that they are non-nil)
- add 1 to the `delegated_class` method's expected argument size to get the argument index
- if that argument index is nil, set it to `{}`, otherwise, leave it alone.
- then, take that argument hash, and set the `client` key.

It assumes a lot, and feels brittle. Have you thought about alternate ways of doing this? `method_missing` may be too "magical" for this purpose.